Sources/MongoSwift/MongoClient.swift

```diff
@@ -148,19 +148,25 @@ public class MongoClient {
 
         if options?.eventMonitoring == true { self.initializeMonitoring() }
 
-        mongoc_client_set_error_api(self._client, MONGOC_ERROR_API_VERSION_2)
+        guard mongoc_client_set_error_api(self._client, MONGOC_ERROR_API_VERSION_2) else {
+            throw RuntimeError.internalError(message: "Could not configure error handling on client")
+        }
     }
 
     /**
      * Create a new client from an existing `mongoc_client_t`.
      * Do not use this initializer unless you know what you are doing.
+     * If this client was derived from a pool, ensure that the error api version was set to 2 on the pool.
      *
      * - Parameters:
      *   - fromPointer: the `mongoc_client_t` to store and use internally
      */
     public init(fromPointer: OpaquePointer) {
         self._client = fromPointer
 
+        // This call may fail, and if it does, either the error api version was already set or the client was derived
+        // from a pool. In either case, the error handling in MongoSwift will be incorrect unless the correct api
+        // version was set by the caller.
```

Comment:
since this initializer is really only for use in MongoMobile, it might be fine for us to assume that whatever is going on there in terms of setting the error API version (or not) will be the case here. 
I don't think there would ever be a pooled client coming from there. I'm not sure if mongoc_embedded is doing anything in terms of setting the error API when we call `mongoc_embedded_v1_client_create` ([here](https://github.com/mongodb/swift-mongo-mobile/blob/64af359f205860fcdba547c698352412f786eb13/Sources/MongoMobile/MongoMobile.swift#L170)) but I would guess not? maybe worth checking.

if it's not being set there, then we might want to handle the error as we do in the other initializer if there are any other possible reasons this could throw.

side note: we should probably mark this initializer with :nodoc: since it's not meant for consumption by anyone but us.

Reply:
marked the initializer with `:nodoc:`. 

After offline discussion, we decided it wasn't worth marking this as `throws` due to the subsequent necessary MongoMobile changes, and since this is only called in MongoMobile, we should know to not be setting the error api version / appropriately handling a pool. So, it is ok that we allow it to fail silently and simply leave a warning comment.